app/models/agents/liquid_output_agent.rb

```diff
@@ -0,0 +1,131 @@
+module Agents
+  class LiquidOutputAgent < Agent
+    include WebRequestConcern
+    include FormConfigurable
+
+    cannot_be_scheduled!
+
+    description  do
+      <<-MD
+        The Liquid Output Agent outputs events through a Liquid template you provide.  Use it to create a HTML page, or a json feed, or anything else that can be rendered as a string from your stream of Huginn data. 
+
+        This Agent will output data at:
+
+        `https://#{ENV['DOMAIN']}#{Rails.application.routes.url_helpers.web_requests_path(agent_id: ':id', user_id: user_id, secret: ':secret', format: :any_extension)}`
+
+        where `:secret` is thel secret specified in your options.  You can use any extension you wish.
+
+        Options:
+
+          * `secrets` - An array of tokens that the requestor must provide for light-weight authentication.
+          * `expected_receive_period_in_days` - How often you expect data to be received by this Agent from other Agents.
+          * `content` - The content to display when someone requests this page.
+          * `mime_type` - The mime type to use when someone requests this page.
+          * `mode` - The behavior that determines what data is passed to the Liquid template.
+
+        # Liquid Templating
+
+        The content you provide will be run as a Liquid template. The data from the last event received will be used when processing the Liquid template.
+
+        # Modes
+
+        ### Merge events
+
+          The data for incoming events will be merged. So if two events come in like this:
+
+```
+{ 'a' => 'b',  'c' => 'd'}
+{ 'a' => 'bb', 'e' => 'f'}
+```
+
+          The final result will be:
+
+```
+{ 'a' => 'bb', 'c' => 'd', 'e' => 'f'}
+```
+
+        This merged version will be passed to the Liquid template.
+
+        ### Last event in
+
+          The data from the last event will be passed to the template.
+      MD
+    end
+
+    def default_options
+      {
+        "secrets" => "a-secret-key",
+        "expected_receive_period_in_days" => 2,
+        "content" => 'This is a Liquid template. Include variables from your last event, like {{this}} and {{that}}.',
+        "mime_type" => 'text/html',
+        "mode" => 'Last event in',
+      }
+    end
+
+    form_configurable :secrets
+    form_configurable :expected_receive_period_in_days
+    form_configurable :content, type: :text
+    form_configurable :mime_type
+    form_configurable :mode, type: :array, values: [ 'Last event in', 'Merge events']
+
+    def working?
+      last_receive_at && last_receive_at > options['expected_receive_period_in_days'].to_i.days.ago && !recent_error_logs?
+    end
+
+    def validate_options
+      if options['secrets'].present?
+        case options['secrets']
```

Comment:
It is a string in this case. It's not in the data output agent because it uses the JSON builder. I was thinking that the form_configurable option was better, but I don't know how to provide a multi-select with the tool. So instead I just limit this to one secret. 

Thoughts? Should this be just one? Or should I drop form_configurable? Or maybe there's another option I'm not seeing?